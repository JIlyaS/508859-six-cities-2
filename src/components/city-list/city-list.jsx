import React from 'react';
import PropTypes from 'prop-types';

const CityList = (props) => {
  const {allOffers, city, changeCityClickHandler} = props;
  const offerCityNames = allOffers.map((offer) => offer.city.name);
  return <ul className="locations__list tabs__list">
    {[...new Set(offerCityNames)].map((offerCity) => <li className="locations__item" key={`location-${offerCity}`}>
      <a
        className={`locations__item-link tabs__item ${city === offerCity && `tabs__item--active`}`}
        href="#"
        onClick={() => changeCityClickHandler(allOffers, offerCity)}
      >
        <span>{offerCity}</span>
      </a>
    </li>)}
  </ul>;
};

CityList.propTypes = {
  city: PropTypes.string.isRequired,
  allOffers: PropTypes.arrayOf(PropTypes.shape({})),
  changeCityClickHandler: PropTypes.func.isRequired
};

export default CityList;
