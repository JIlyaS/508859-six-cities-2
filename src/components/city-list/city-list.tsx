import * as React from 'react';
import {connect} from 'react-redux';

import ActionCreator from '../../action-creator/action-creator';
import {CityListProps} from '../../types/types';

const CityList: React.FunctionComponent<CityListProps> = (props) => {
  const {city, cities, onChangedCityClick} = props;
  return <ul className="locations__list tabs__list">
    {[...new Set(cities)].map((offerCity) => <li className="locations__item" key={`location-${offerCity}`}>
      <a
        className={`locations__item-link tabs__item ${city === offerCity && `tabs__item--active`}`}
        href="#"
        onClick={() => onChangedCityClick(offerCity)}
      >
        <span>{offerCity}</span>
      </a>
    </li>)}
  </ul>;
};

const mapStateToProps = (state, ownProps) =>
  Object.assign({}, ownProps, {
    city: state.userReducer.city,
    cities: state.appReducer.offers.map((offer) => offer.city.name)
  });

const mapDispatchToProps = (dispatch) => ({
  onChangedCityClick: (city) => dispatch(ActionCreator.changeCity(city))
});

export {CityList};

export default connect(mapStateToProps, mapDispatchToProps)(CityList);
