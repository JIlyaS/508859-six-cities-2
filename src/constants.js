export const RATING_PERSENT = 100;
export const MAX_RATING_COUNT = 5;
export const ICON = {
  url: `img/pin.svg`,
  size: [30, 30]
};
export const LEAFLET_OPTION = {
  center: [52.38333, 4.9],
  zoom: 12
};
export const TILE_LAYER = {
  img: `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
  attributions: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
};
