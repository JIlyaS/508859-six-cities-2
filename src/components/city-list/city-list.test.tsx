import * as React from 'react';
import * as Enzyme from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as Adapter from 'enzyme-adapter-react-16';

import {CityListMock} from '../../constants';
import {CityList} from './city-list';

Enzyme.configure({adapter: new Adapter()});

it(`CityList correctly renders after relaunch`, () => {
  const tree = Enzyme.shallow(
      <CityList
        city={CityListMock.CITY}
        cities={CityListMock.CITIES}
        onChangedCityClick={() => {}}
      />
  );

  expect(toJSON(tree)).toMatchSnapshot();
});
