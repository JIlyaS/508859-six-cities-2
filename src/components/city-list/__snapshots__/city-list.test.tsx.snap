// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityList correctly renders after relaunch 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
    key="location-Amsterdam"
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
      href="#"
      onClick={[Function]}
    >
      <span>
        Amsterdam
      </span>
    </a>
  </li>
  <li
    className="locations__item"
    key="location-Paris"
  >
    <a
      className="locations__item-link tabs__item false"
      href="#"
      onClick={[Function]}
    >
      <span>
        Paris
      </span>
    </a>
  </li>
</ul>
`;
